src/elements/content-sidebar/__tests__/withSidebarAnnotations.test.js

```diff
@@ -0,0 +1,267 @@
+        feedItems: jest.fn(),
+        getCachedItems: jest.fn(),
+        deleteAnnotation: jest.fn(),
+    };
+
+    const api = {
+        getFeedAPI: () => feedAPI,
+    };
+
+    const defaultProps = { api, ...annotatorContextProps, file };
+
+    const getWrapper = props => shallow(<WrappedComponent {...defaultProps} {...props} />);
+
+    describe('constructor', () => {
+        test('should call redirectDeeplinkedAnnotation', () => {
+            getWrapper();
+
+            expect(annotatorContextProps.getAnnotationsMatchPath).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    describe('componentDidUpdate', () => {
+        test.each`
+            prevFileVersionId | fileVersionId | expectedCallCount
+            ${'122'}          | ${'122'}      | ${0}
+            ${'122'}          | ${'123'}      | ${1}
+        `(
+            'should call updateActiveVersion if fileVersionId changes',
+            ({ prevFileVersionId, fileVersionId, expectedCallCount }) => {
+                const match = { params: { fileVersionId } };
+                const prevMatch = { params: { fileVersionId: prevFileVersionId } };
+                const wrapper = getWrapper({ location: 'foo' });
+                const instance = wrapper.instance();
+
+                instance.updateActiveVersion = jest.fn();
+                annotatorContextProps.getAnnotationsMatchPath.mockReturnValueOnce(match).mockReturnValueOnce(prevMatch);
+
+                wrapper.setProps({ location: 'bar' });
+                expect(instance.updateActiveVersion).toHaveBeenCalledTimes(expectedCallCount);
+            },
+        );
+
+        test.each`
+            condition                                          | prevActiveAnnotationId | activeAnnotationId | isAnnotationsPath | expectedCount
+            ${'annotation ids are the same'}                   | ${'123'}               | ${'123'}           | ${true}           | ${0}
+            ${'annotation ids are different'}                  | ${'123'}               | ${'456'}           | ${true}           | ${1}
+            ${'annotation deselected on annotations path'}     | ${'123'}               | ${null}            | ${true}           | ${1}
+            ${'annotation deselected not on annotations path'} | ${'123'}               | ${null}            | ${false}          | ${0}
+            ${'annotation selected not on annotations path'}   | ${null}                | ${'123'}           | ${false}          | ${1}
+            ${'annotation selected on annotations path'}       | ${null}                | ${'123'}           | ${true}           | ${1}
+        `(
+            'should call updateActiveAnnotation $expectedCount times if $condition',
+            ({ prevActiveAnnotationId, activeAnnotationId, isAnnotationsPath, expectedCount }) => {
+                const wrapper = getWrapper({ annotatorState: { activeAnnotationId: prevActiveAnnotationId } });
+
+                wrapper.instance().updateActiveAnnotation = jest.fn();
```

Comment:
I've been favoring `jest.spyOn` for instance methods to keep from overwriting the property.